doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1257,409 +1261,344 @@ Managing the local context
-   This lets you avoid specifying the asserted statement completely before starting
-   to prove it.
+   .. tacn:: eassert ( @ident : @type ) {? by @ltac_expr3 }
+             eassert ( @ident := @term )
+             eassert @one_term {? @as_ipat } {? by @ltac_expr3 }
+      :name: eassert; _; _
 
-.. tacv:: pose proof @term {? as @simple_intropattern}
-   :name: pose proof
+      Unlike :tacn:`assert`, the :n:`@type`\s, :n:`@term`\s and :n:`@one_term`\s in
+      :tacn:`eassert` may contain :gdef:`holes <hole>`, denoted by :n:`_`,
+      for which the tactic will create existential variables.  This lets you
+      avoid specifying the asserted statement completely before starting to
+      prove it.
 
-   This tactic behaves like :n:`assert @type {? as @simple_intropattern} by exact @term`
-   where :token:`type` is the type of :token:`term`. In particular,
-   :n:`pose proof @term as @ident` behaves as :n:`assert (@ident := @term)`
-   and :n:`pose proof @term as @simple_intropattern` is the same as applying the
-   :token:`simple_intropattern` to :token:`term`.
+.. tacn:: enough ( @ident : @type ) {? by @ltac_expr3 }
+          enough @one_term {? @as_ipat } {? by @ltac_expr3 }
+   :name: enough; _
 
-.. tacv:: epose proof @term {? as @simple_intropattern}
-   :name: epose proof
+   Adds a new hypothesis to the current subgoal and a new subgoal to prove the
+   hypothesis.
 
-   While :tacn:`pose proof` expects that no existential variables are generated by
-   the tactic, :tacn:`epose proof` removes this constraint.
+   The first form adds a new hypothesis :n:`@ident : @type`
+   and :n:`@type` as the new subgoal.  Then, if :n:`@ltac_expr3` is
+   specified, it applies that tactic to prove the current subgoal
+   (and otherwise fails).
 
-.. tacv:: pose proof (@ident := @term)
+   In the second form, if :n:`@as_ipat` isn't specified, the tactic adds a new
+   hypothesis :n:`@one_term` with a name chosen by Coq.  Otherwise, it transforms
+   :n:`@one_term` as specified by :n:`@as_ipat` and adds the resulting new hypotheses.
+   The :n:`@as_ipat` may also expand the current subgoal into multiple subgoals.
+   Then, if :n:`@ltac_expr3` is specified, it is applied to and must succeed on all
+   of them.
 
-   This is an alternative syntax for :n:`assert (@ident := @term)` and
-   :n:`pose proof @term as @ident`, following the model of :n:`pose
-   (@ident := @term)` but dropping the value of :token:`ident`.
+   .. tacn:: eenough ( @ident : @type ) {? by @ltac_expr3 }
+             eenough @one_term {? @as_ipat } {? by @ltac_expr3 }
+      :name: eenough; _
 
-.. tacv:: epose proof (@ident := @term)
+      Unlike :tacn:`enough`, the :n:`@type`\s and :n:`@one_term`\s in
+      :tacn:`eenough` may contain :term:`holes <hole>`, denoted by :n:`_`,
+      for which the tactic will create existential variables.  This lets you
+      avoid specifying the asserted statement completely until you later
+      start to prove it.
```

Comment:
In fact, the holes in the asserted statement may very well be resolved before we start to prove it by the use we make of it, so what about:
```suggestion
      avoid specifying the asserted statement completely until you start
      to use it or later start to prove it.
```

Reply:
the "it" you suggest adding refers to the statement or the existential variable?

I don't know what specifically "use" refers to.  Can we be a little more specific without too much discussion?

Reply:
I meant the statement, but to be more specific, the added hypothesis of the asserted statement.

Does that make clearer what "use" refers to?

Reply:
I put this:

This lets you avoid specifying the asserted statement completely until you start to use
the hypothesis or later start to prove the statement.

I'm not sure _how_ you start using the hypothesis, but I won't concern myself with that for now.  Ignorance is bliss :-)

Reply:
> I'm not sure _how_ you start using the hypothesis

With a tactic, like `apply` or `rewrite`.